src/avfoundation.cs

```diff
@@ -9444,10 +9462,23 @@ interface AVCaptureDevice {
 		[Static, Export ("authorizationStatusForMediaType:")]
 		AVAuthorizationStatus GetAuthorizationStatus (NSString avMediaTypeToken);
 
+		// Calling this method with any media type other than AVMediaTypeVideo or AVMediaTypeAudio raises an exception.
```

Comment:
I believe this comment should either become an `[Advice]` or be documented to avoid unexpected behaviors. I see that you default to `Audio` if `AVAuthorizationMediaType` is anything else than `Video` but we need to inform the user.

Reply:
Not needed, the only two possible options in the enum are 

```
enum AVAuthorizationMediaType {
 	Video,
 	Audio,
 }
```

Reply:
Ah misread I thought it was `enum AVMediaTypes`.

Not sure why you added the comment then (;

Reply:
@VincentDondain to justify the `AVAuthorizationMediaType` enum having just two options whereas `AVMediaTypes` enum having a lot more options, maybe my comment must had been clearer. 😃

Reply:
But then what if Apple add more `AVAuthorizationMediaType` values?

You wrapper is just preventive?

Reply:
You would just add values to the existing `AVAuthorizationMediaType` enum and do the respective check in `AVAuthorizationStatus GetAuthorizationStatus (AVAuthorizationMediaType)` to use the right key. 

Or you could just use `AVAuthorizationStatus GetAuthorizationStatus (NSString);` directly since it is already there.

Reply:
Yea and I missed the fact the wrapper is not preventive it allows us to pass `AVMediaTypes.Video.GetConstant ()` for `AVAuthorizationMediaType.Video`, etc...

This could get complicated very quickly though (: